egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -20,14 +20,14 @@ def GetArgs():
                                                  "--random-seed 1 data/train data/train_rvb",
                                      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-    parser.add_argument("--rir-list-file", type=str, required = True, 
+    parser.add_argument("--rir-list-file", type=str, action='append', required = True, dest = "rir_list_file_array", 
                         help="RIR information file, the format of the file is "
                         "--rir-id <string,required> --room-id <string,required> "
                         "--receiver-position-id <string,optional> --source-position-id <string,optional> "
                         "--rt-60 <float,optional> --drr <float, optional> location <rspecifier> "
                         "E.g. --rir-id 00001 --room-id 001 --receiver-position-id 001 --source-position-id 00001 "
                         "--rt60 0.58 --drr -4.885 data/impulses/Room001-00001.wav")
-    parser.add_argument("--noise-list-file", type=str, default = None,
+    parser.add_argument("--noise-list-file", type=str, action='append', default = None, dest = "noise_list_file_array",
```

Comment:
I would suggest something like
--rir-list-parameters "0.7,<filrname>"
--rir-list-parameters "0.3,<filrname>"

If no value is provided for some lists you just divide the remaining
probability mass among them.

If you want you could change the option name to be more descriptive, but I
would prefer keeping the filename and probability value together to prevent
user errors when several lists are being given. I finally hope to use this
script with ~10 different RIR databases we have access to, so this would be
a convenient format in that case.

Vijay

Vijay

On Aug 24, 2016 05:32, "tomkocse" <notifications@github.com> wrote:

> In egs/wsj/s5/steps/data/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/992#discussion_r76024383>:
>
> >                          help="RIR information file, the format of the file is "
> >                          "--rir-id <string,required> --room-id <string,required> "
> >                          "--receiver-position-id <string,optional> --source-position-id <string,optional> "
> >                          "--rt-60 <float,optional> --drr <float, optional> location <rspecifier> "
> >                          "E.g. --rir-id 00001 --room-id 001 --receiver-position-id 001 --source-position-id 00001 "
> >                          "--rt60 0.58 --drr -4.885 data/impulses/Room001-00001.wav")
> > -    parser.add_argument("--noise-list-file", type=str, default = None,
> > +    parser.add_argument("--noise-list-file", type=str, action='append', default = None, dest = "noise_list_file_array",
>
> @vijayaditya <https://github.com/vijayaditya> What is the user interface
> you prefer, what about adding an option like --rir-mass-probabilities
> "0.3:0.7" ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/992/files/680a20cd2c4f297d9edc4ffdf386a28586ba4ef8#r76024383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADtwoBEZWQa7jeMP4AmQt-Y0ywOYrAspks5qjA-cgaJpZM4JpoP9>
> .
>